### ISSUE TITLE
Allow properties to be added to Geo types

### ISSUE THREAD

**Issue by jerel** [bryanjos/geo#74]
Is there a PostGIS (or other) limitation that would prevent us from adding `properties` to the Geo structs?

For example it would be very nice to be able to do the following:

```
%Geo.Point{coordinates: {1, 1}, properties: %{time: "12:01"}}
|> Geo.JSON.encode()
```

**Comment by bryanjos** [comment]
There is no limitation that would prevent that. It would just be ignored unless used by something like the Geo.JSON module. I think adding properties is fine. 

If another request for adding a GeoJSON specific item comes up, I'm leaning towards making something that follows the spec exactly that wraps around the Geo structs.

**Comment by larsbutler** [comment]
@jerel 

> Is there a PostGIS (or other) limitation that would prevent us from adding properties to the Geo structs?

The [GeoJSON RFC states](https://tools.ietf.org/html/rfc7946#section-7.1) that "FeatureCollection and Geometry objects, respectively, MUST NOT contain a "geometry" or "properties" member." This came up in a [discussion I had with one of the RFC authors](https://github.com/geomet/geomet/issues/28#issuecomment-261363251). 

It seems that the general theme here is a lack of support for metadata. Your cases is pretty custom, but there are some cases where metadata is vital to retaining information when converting between formats (GeoJSON to/from EWKT/EWKB, for example). Relevant issues:

- https://github.com/geomet/geomet/issues/28
- https://github.com/bryanjos/geo/issues/76

I would really like to see some official guidance for how to include metadata/extensions in the GeoJSON schema. I'll have to go and do a bit of research.

**Comment by niahoo** [comment]
Hi,

That's exactly what the `Feature` geojson type is for. It has two members, `geometry` and `properties`.

`FeatureCollection` has only one member (besides `type`) : `features :: [Feature]`.

Any plans to implement it ?

**Comment by bryanjos** [comment]
I've been thinking about this a lot and I think instead of trying to shoehorn geojson features into the geometry types, maybe it would be cleaner to create struct(s) around geojson itself. Maybe that's a better direction to move towards

**Comment by niahoo** [comment]
But this is not stoehorning, the Feature type exists for this sole purpose : have geometry and properties in one entity ...

**Comment by bryanjos** [comment]
Looking at the spec and thinking on it a bit, I think adding `properties` to the types should be fine.

**Comment by bryanjos** [comment]
Fixed in 4647ed357a0ffdb3376c64204d9b79521caef67e
